revolv/project/views.py

```diff
@@ -62,15 +77,28 @@ def get_success_url(self):
         messages.success(self.request, 'Project details updated')
         return reverse('project:view', kwargs={'pk': self.get_object().id})
 
+    # validates project, formset of donation levels, and adds categories as well
     def form_valid(self, form):
-        project = self.get_object()
-        project.update_categories(form.cleaned_data['categories_select'])
+        context = self.get_context_data()
```

Comment:
Can we implement like a `get_formset` method for this? It's fairly weird to be calling `get_context_data` in a POST request